### ISSUE TITLE
Connections Security Preferences

### ISSUE THREAD

**Issue by jcjones37** [SparkDevNetwork/Rock#4589]
### Description
When setting security on a Connection Opportunity, the [Connection Requests] block on a _person profile page_ seems to ignore the security settings. However, the security setting does work as expected from the [My Connection Opportunities] block on the _connections page_.

### Steps to Reproduce
1: Go to the connections page.
2: Click on the settings cog, to reach the Connections Type page.
3: Go into a Connections Type
4: Select a Connection Opportunity and modify the security settings.
5: Change the View Security of that Connection Opportunity to a specific group of people (say rock admins) and restrict all other users.
6: Create a connection of that type and that opportunity for someone in the database.
7: Notice as a Rock Admin, that you can see that opportunity block on the Connections Page within the [My Connection Opportunities] block.
8: Now, as a Rock Admin, go to the persons profile page that you created a connection for, and notice that you see that opportunity link within the [Connection Requests] block on their profile.
9: Impersonate someone who has access to rock but is not a Rock Admin.
10:  Notice as someone who is not a Rock Admin, that you can not see that opportunity block on the Connections Page within the [My Connection Opportunities] block. - NICE
11: Now, as someone who is not a Rock Admin, go to the persons profile page that you created a connection for, and notice that you see that opportunity link within the [Connection Requests] block on their profile. - BUMMER

**Expected behavior:**

I would expect if I restricted a connection opportunity, any connections of that opportunity would not be visible to people without those security roles from the Connections Page or the Person Profile Page (two different blocks)

**Actual behavior:**

When a particular connection opportunity is secured, the connections page [My Connections Opportunities] block does restrict viewing access to that opportunity, the opportunity still shows on peoples profile who have a connection associated with that connection opportunity (even if your permission does not allow for view rights).

### Versions
* **Rock Version:** [11.3]
* **Client Culture Setting:** [en-US]

**Comment by bengoshow** [comment]
I'll second this - I need this functionality for a client site and I can reproduce it on the demo site.

I added a Security setting to the View tab for the Greeter connection opportunity. It's hidden now on the back-end when you edit the Connection Type https://rock.rocksolidchurchdemo.com/page/410?ConnectionTypeId=1 (because Alisha isn't in the RSR - Connections Admin role), but still visible on the front end at https://rock.rocksolidchurchdemo.com/page/235, and also still visible in the Connections list at https://rock.rocksolidchurchdemo.com/page/2634

![Screen Shot 2021-04-02 at 10 04 47 AM](https://user-images.githubusercontent.com/1636930/113422708-2b43a700-939b-11eb-94b4-5cd2d946978b.png)

![Screen Shot 2021-04-02 at 10 04 19 AM](https://user-images.githubusercontent.com/1636930/113422720-30085b00-939b-11eb-93d6-e8ff09cd81d3.png)

![Screen Shot 2021-04-02 at 10 05 10 AM](https://user-images.githubusercontent.com/1636930/113422724-339be200-939b-11eb-95d2-547062cc53a7.png)

**Comment by jonedmiston** [comment]
@bengoshow you're comment is referring to a different issue. When we wrote the external search 'security' on opportunities was not a feature we supported. The intended use-case was for searching for opportunities within a specific public connection type.

We're adding some other security fixes/changes, so if you add it as a separate issue we'll add it to the list. It would be helpful for us if you listed the church you are working with that has the need of this.
